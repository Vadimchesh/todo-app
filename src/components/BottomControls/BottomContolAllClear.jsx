import React from 'react';
import styles from './style.module.scss';
import { useDispatch } from 'react-redux';

const BottomContolAllClear = (id) => {
  const dispatch = useDispatch();

  const onClearCompletedClicked = () =>
    dispatch({ type: 'todos/completedClear', payload: id });

  return (
    <label className={styles.label}>
      <span id='clear' onClick={onClearCompletedClicked}>
        Clear all complited
      </span>
    </label>
  );
};

export default BottomContolAllClear;
