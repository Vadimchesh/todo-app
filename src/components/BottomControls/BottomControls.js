import React from 'react';
import { StatusFilters } from '../../redux/reducerFilter';
import styles from './style.module.scss';

const BottomControls = ({ value: status, onChange }) => {
  const renderedFilters = Object.keys(StatusFilters).map((key) => {
    const value = StatusFilters[key];
    const handleClick = () => onChange(value);

    return (
      <label className={styles.label} key={value}>
        <input type='radio' name='bottom' />
        <span onClick={handleClick}>{key}</span>
      </label>
    );
  });

  return <div className='styles.conainer'>{renderedFilters}</div>;
};

export default BottomControls;
