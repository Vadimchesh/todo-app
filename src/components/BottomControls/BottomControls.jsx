import React from "react";
import FilterStatus from "../FilterStatus/FilterStatus";
import styles from "./style.module.scss";

function BottomControls() {
  return (
    <div className={styles.container}>
      <label>2 2 left</label>
      <FilterStatus />
      <label>Clear completed</label>
    </div>
  );
}

export default BottomControls;
