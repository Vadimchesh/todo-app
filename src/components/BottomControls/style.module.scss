@import "../../assets/scss/variables.scss";

.container {
  display: flex;
  justify-content: space-between;
  padding: 1.5rem;
  font-size: 1.2rem;
}

.conainer {
  padding: 0.5rem;
  cursor: pointer;
}
.label {
  cursor: pointer;
  padding: 0 1rem;
}
.label input {
  display: none;
  padding: 1rem;

  &:checked + span {
    padding: 0.2rem;
    border: 1px solid $red-color;
  }
}
