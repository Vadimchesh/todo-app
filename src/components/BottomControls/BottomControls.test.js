import React from 'react';
import BottomControls from './BottomControls';
import renderer from 'react-test-renderer';
import { shallow, ShallowWrapper } from 'enzyme';
import { useDispatch } from 'react-redux';

const mockDispatch = jest.fn();

jest.mock('react-redux', () => {
  const originalModule = jest.requireActual('react-redux');
  return {
    ...originalModule,
    useDispatch: jest.fn(() => mockDispatch),
  };
});

describe('BottomControls:', () => {
  let wrapper = ShallowWrapper;
  beforeEach(() => {
    wrapper = shallow(<BottomControls />);
  });
  afterAll(() => {
    jest.resetAllMocks();
  });

  it('should render all label', () => {
    const label = wrapper.find('label');
    expect(label.length).toBe(3);
  });

  it('renders correctly', () => {
    const tree = renderer.create(<BottomControls />);
    expect(tree).toMatchSnapshot();
  });
});
