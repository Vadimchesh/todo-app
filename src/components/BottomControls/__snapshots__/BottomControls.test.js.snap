// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BottomControls: renders correctly 1`] = `
<div
  className="styles.conainer"
>
  <label
    className="label"
  >
    <input
      name="bottom"
      type="radio"
    />
    <span
      onClick={[Function]}
    >
      All
    </span>
  </label>
  <label
    className="label"
  >
    <input
      name="bottom"
      type="radio"
    />
    <span
      onClick={[Function]}
    >
      Active
    </span>
  </label>
  <label
    className="label"
  >
    <input
      name="bottom"
      type="radio"
    />
    <span
      onClick={[Function]}
    >
      Completed
    </span>
  </label>
</div>
`;
